app/models/spree/credit_card_decorator.rb

```diff
@@ -14,6 +14,9 @@
 
   belongs_to :user
 
+  after_create :ensure_default
```

Comment:
isn't this callback then redundant? `after_save` will also be called on creation.

Reply:
The trick is is that `is_default_changed?` doesn't seem to return true on create, so this was the best way I could think of to achieve what I needed: always run on create, and then any time the value is changed.